src/solver/hydro/management/management.cpp

```diff
@@ -219,38 +229,43 @@ void HydroManagement::prepareEffectiveDemand(uint numSpace)
             assert(not Math::NaN(data.MLE[realmonth]) && "nan value detected for DLE");
         }
 
-        auto minimumYear = std::numeric_limits<double>::infinity();
-        auto dayYear = 0u;
-
-        for (uint month = 0; month != 12; ++month)
+        if (parameters.hydroAllocationHandleNegativeDemand
```

Comment:
I would recommend a small refactoring for function  **prepareEffectiveDemand** : 
```C++
bool HydroManagement::negativeEffeciveDemandsRequireIncreasing()
{
    return parameters.handleNegativeDemand == increasingEffectiveDemands;
}

...

void HydroManagement::prepareEffectiveDemand(uint numSpace)
{
    this->computeEffectiveDemandsFromAllocationMatrix(numSpace)
    if (this->negativeEffeciveDemandsRequireIncreasing())
    {
        this->increaseEffectiveDemands()
    }
}
```